### ISSUE TITLE
plot.FFTrees() is not written in a modular way and should be refactored

### ISSUE THREAD

**Issue by ndphillips** [ndphillips/FFTrees#221]
Currently `plot.FFTrees()`, creates a dashboard of multiple plotting elements (i.e.; Bars, tree, icons, confusion matrix, ROC). 

<img width="507" alt="image" src="https://github.com/ndphillips/FFTrees/assets/8480980/d9ed31a4-a3ca-4cbd-a951-133e71095eda">

The script creating this dashboard https://github.com/ndphillips/FFTrees/blob/master/R/plotFFTrees_function.R is over 2,000 lines long in one massive function.

This makes it difficult to debug and doesn't lend itself well to creating user functions that allow them to selectively plot _some_ of the plotting elements, some of which may not even technically need data to create.

For example, I have often wanted to quickly plot an example FFT, without any icons, for demonstration purposes and without the need to actually train on data - just to create a diagram representing a tree of interest. Currently the code to do this is in the function but can't easily be separated from training a tree.

Goals
* Re-factor `plot.FFTrees()` to be a wrapper around sub-functions that create each of the elements of the dashboard.

**Comment by hneth** [comment]
Yes, that's a great idea.  And it would be great if each component of the complex panel (e.g., the baseline data info, icon arrays, or 2x2 matrix) could be plotted separately as well (as it currently works for the ROC curve).  But given the length of the current function and its complex dependencies (for the x-y-coordinates), I've so far shied away from this challenge — so I'd be delighted if you could take it on!

**Comment by ndphillips** [comment]
Thanks glad to see you agree.

Fixing this is indeed quite a challenge. The code underlying `plot.FFTrees()` is as poorly designed as the output is beautiful :) -- I'm ok saying that out loud since it's all my fault!

**Comment by ndphillips** [comment]
Let me try breaking down the plot into its elements:

## Core Plotting Elements

1.  Level Bar
1.  Icon Array
1.  Tree
1.  Confusion Matrix
1.  ROC

## Titles and Labels
1. Section Titles
1. Icon labels

## Layout
1. Overall Layout

**Comment by ndphillips** [comment]
From this, I would propose creating the following functions:

|Function|Element|
|:------|:------|
|`plot_icon_label()`| L1|
|`plot_level_bar()`| G1|
|`plot_icon_array()`|G2|
|`plot_tree()`|G3|
|`plot_confusion_matrix()`|G4|
|`plot_roc()`| G5|

**Comment by ndphillips** [comment]
One big question I do have is whether to transition to `ggplot2` or stay in base-R.

I love ggplot2. In my personal and professional work, I have completely transitioned to it.

Pros
* Flexible - ggplot2 is super flexible and extendable. Opens up the possibilities of introducing elements such as animations (https://gganimate.com/), themes ([gge](https://ggplot2.tidyverse.org/reference/ggtheme.html), and interactivity (https://plotly.com/ggplot2/)

Cons
* Time - Will basically require re-writing all plotting functionality from scratch
* Dependencies - Will increase number of dependencies in package

**Comment by hneth** [comment]
**Thanks** for the breakdown in modular components, which is very helpful.  When re-creating the summary plot from these components, it would be much easier to create alternative versions or swap individual components.  

For instance, I've thinking about replacing the **icon arrays** at the exits with **2x2 matrices** that sum up either the classified cases at the current node (i.e., either the top or bottom row of a 2x2 matrix) OR all classified cases so far.  Additionally, it would be helpful to visualize how **NA values** in cues are being handled (e.g., show their number at nodes and how they are classified eventually).  Adding such functionality in a more modular framework may involve editing several functions, but seems far easier than fiddling with the complex monolithic function we had before. 

With regards to **ggplot2**:  I love and use this package as well, but see more immediate costs than benefits by transitioning to it.  I we were to develop a general infrastructure for plotting trees and aspects of their elements, **ggplot2** may be an option (especially if we wanted a wider community to contribute).  But as FFTs have some special constraints and our present plotting options are still quite sophisticated I'd opt for a **base** R solution within **FFTrees**.

**Comment by ndphillips** [comment]
Thanks Hans. I agree that, at least to start, sticking with  base-R is a reasonable option. I'm working on this in #224

**Comment by hneth** [comment]
Excellent!  btw: I previously wrote a bunch of plotting macros (e.g., functions for drawing boxes and links between them) for the diagrams in the **riskyr** package (see [plot_util.R](https://github.com/hneth/riskyr/blob/master/R/plot_util.R)).  Some of these functions are too specific for **FFTrees**, but the more basic ones could be helpful when drawing trees.
